src/services/CovidIndiaApiClient/index.ts

```diff
@@ -1,8 +1,85 @@
+import "reflect-metadata";
 import { ICovidIndiaApiClient } from "../../interfaces/covid-india-api-client.interface";
-import { injectable } from "inversify";
+import { injectable, inject } from "inversify";
+import { WebhookClient } from "dialogflow-fulfillment";
+import axios, { AxiosResponse } from 'axios';
+import { COVID_INDIA, INDIA_PINCODE_API } from "../../constants/host-names";
+import { pinCodeNotFound, areaNotFoundInData, reply } from "./messages.contants";
+import { DependencyIdentifier } from "../../DependencyIdentifiers";
+import { IMessage } from "../../interfaces/messages.iterface";
+
 @injectable()
 export class CovidIndiaApiClient implements ICovidIndiaApiClient {
-    getCovidData() {
-        console.log('get the data here')
+    constructor(@inject(DependencyIdentifier.MESSAGES) private messages: IMessage) { }
+    public async getStatsByPincode(agent: any): Promise<void> {
+        // Getting the the district name and the state name using post code api
+        // Also getting the corona data from india api
+        const response: any[] = await Promise.all([
+            this.getPostCodeDetails(agent.parameters.pincode),
+            this.getDataFromApi()
+        ])
+
+        const pinResponse = response[0];
+        const dataResponse = response[1];
```

Comment:
Suggestion: We can use array destructing. const [pinResponse, dataResponse]  = await Promise.all